g2core/spindle.cpp

```diff
@@ -369,6 +375,23 @@ void spindle_end_override(const float ramp_time)
     return;
 }
 
+// Called in ISR so be careful and fast
+void spindle_update_laser_override(float current_velocity) {
+    float spindle_override = 0.0;
+    float feed_rate = cm_get_feed_rate(ACTIVE_MODEL);
+
+    if (feed_rate > 0.0) {
+        spindle_override = current_velocity / feed_rate;
+        if (spindle_override > 1.0) spindle_override = 1.0;
+        else if (spindle_override < 0.0) spindle_override = 0.0; 
+    }
+//    if (fabs(spindle_override - spindle.override_factor) >= 0.01) {
+        // Only update if the change is worth while (at least 1%)
+        spindle.override_factor = spindle_override;
+        pwm_set_duty(PWM_1, _get_spindle_pwm(spindle, pwm));
```

Comment:
Note: This is where I would adjust the i2c code if we were controlling a digital pot for plasma power settings.